src/pybind/mgr/cephadm/services/mgmt_gateway.py

```diff
@@ -41,36 +68,38 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
         self.mgr.set_module_option_ex('dashboard', 'standby_error_status_code', '503')
         self.mgr.set_module_option_ex('dashboard', 'standby_behaviour', 'error')
 
-    def get_certificates(self, svc_spec: MgmtGatewaySpec, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[str, str, str, str]:
-        self.ssl_certs = SSLCerts()
-        old_cert = self.mgr.cert_key_store.get_cert('mgmt_gw_root_cert')
-        old_key = self.mgr.cert_key_store.get_key('mgmt_gw_root_key')
-        if old_cert and old_key:
-            self.ssl_certs.load_root_credentials(old_cert, old_key)
-        else:
-            self.ssl_certs.generate_root_cert(self.mgr.get_mgr_ip())
-            self.mgr.cert_key_store.save_cert('mgmt_gw_root_cert', self.ssl_certs.get_root_cert())
-            self.mgr.cert_key_store.save_key('mgmt_gw_root_key', self.ssl_certs.get_root_key())
-
+    def get_external_certificates(self, svc_spec: MgmtGatewaySpec, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[str, str]:
+        cert = self.mgr.cert_key_store.get_cert('mgmt_gw_cert')
+        key = self.mgr.cert_key_store.get_key('mgmt_gw_key')
+        if not (cert and key):
+            # not available on store, check if provided on the spec
+            if svc_spec.ssl_certificate and svc_spec.ssl_certificate_key:
+                cert = svc_spec.ssl_certificate
+                key = svc_spec.ssl_certificate_key
+            else:
+                # not provided on the spec, let's generate self-sigend certificates
+                addr = self.mgr.inventory.get_addr(daemon_spec.host)
+                host_fqdn = self._inventory_get_fqdn(daemon_spec.host)
+                cert, key = self.mgr.cert_mgr.generate_cert(host_fqdn, addr)
+            # save certificates
+            if cert and key:
+                self.mgr.cert_key_store.save_cert('mgmt_gw_cert', cert)
+                self.mgr.cert_key_store.save_key('mgmt_gw_key', key)
+            else:
+                logger.error("Failed to obtain certificate and key fro mgmt-gateway.")
```

Comment:
```suggestion
                logger.error("Failed to obtain certificate and key from mgmt-gateway.")
```

Reply:
Also, should we ever actually be getting here without a cert and key when just generating one is one of the options?

Reply:
TBH I don't thik we should ever reach this log, but I put it just in case some previous code breaks due to some change so at least we get the error log